portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -0,0 +1,134 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.data.UserEntry;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+/** 
+* Servlet that handles user's comment data 
+*/
+@WebServlet("/data")
+public final class DataServlet extends HttpServlet {
+  // Set default to 5
+  int maxCommentsPosted = 5;
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Gather all the user comments and sort them by timestamp
+    Query query = new Query("Comments").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    List<UserEntry> commentsRecord = collectEntriesToPost(results);
+   
+	// Create commentsRecord object in json form
+    String json = new Gson().toJson(commentsRecord);
+    
+    // Send the JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  /* 
+  * Create the list of comments to be posted on website
+  */
+  private List collectEntriesToPost(PreparedQuery results){
+    // Create list of entities that contain UserEntry objects
+    List<UserEntry> commentsRecord = new ArrayList<>();
+
+    int count = 0;
+    /**
+    * For loop through all the stored results
+    * Once you've gathered the info of the current result check add it to the userEntry
+    * to be posted on website until they reach the requested number of comments
+    */
+    for (Entity entity : results.asIterable()) {
+      count++;
+      long id = entity.getKey().getId();
+      String username = (String) entity.getProperty("userName");
+      String comment = (String) entity.getProperty("userComment");
+      long timestamp = (long) entity.getProperty("timestamp");
+      // Only add to the list if you have yet to reach the number of comments requested 
+      if(count <= maxCommentsPosted){
+        UserEntry userEntry = new UserEntry(id, username, comment, timestamp);
+      	commentsRecord.add(userEntry);
+      }
+  	}
+  }
```

Comment:
You can use this https://stackoverflow.com/questions/29876803/google-app-engine-java-datastore-query-limit-results to limit the count of comments rather than implementing this.